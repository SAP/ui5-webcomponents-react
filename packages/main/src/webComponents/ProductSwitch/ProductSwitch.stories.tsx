import type { Meta, StoryObj } from '@storybook/react';
import businessObjectsExperienceIcon from '@ui5/webcomponents-icons/dist/business-objects-experience.js';
import contactsIcon from '@ui5/webcomponents-icons/dist/contacts.js';
import flightIcon from '@ui5/webcomponents-icons/dist/flight.js';
import homeIcon from '@ui5/webcomponents-icons/dist/home.js';
import wrenchIcon from '@ui5/webcomponents-icons/dist/wrench.js';
import { ProductSwitchItem } from '../ProductSwitchItem/index.js';
import { ProductSwitch } from './index.js';

const meta = {
  title: 'Inputs / ProductSwitch',
  component: ProductSwitch,
  argTypes: {
    children: { control: { disable: true } }
  }
} satisfies Meta<typeof ProductSwitch>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  render: (args) => {
    return (
      <ProductSwitch {...args}>
        <ProductSwitchItem
          titleText={'ProductSwitchItem'}
          icon={wrenchIcon}
          target="_blank"
          targetSrc="https://sap.github.io/ui5-webcomponents-react/"
        />
        <ProductSwitchItem titleText="Home" subtitleText="Central Home" icon={homeIcon} />
        <ProductSwitchItem
          titleText="Analytical Cloud"
          subtitleText="Analytical Cloud"
          icon={businessObjectsExperienceIcon}
        />
        <ProductSwitchItem titleText="Catalog" subtitleText="Ariba" icon={contactsIcon} />
        <ProductSwitchItem titleText="Travel & Expense" subtitleText="Concur" icon={flightIcon} />
      </ProductSwitch>
    );
  }
};
