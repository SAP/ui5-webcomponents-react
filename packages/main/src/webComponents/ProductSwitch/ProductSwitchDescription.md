## Keyboard Handling

The `ProductSwitch` provides advanced keyboard handling. When focused, the user can use the following keyboard shortcuts in order to perform a navigation:  

*   \[TAB\] - Move focus to the next interactive element after the `ProductSwitch`
*   \[UP/DOWN\] - Navigates up and down the items
*   \[LEFT/RIGHT\] - Navigates left and right the items