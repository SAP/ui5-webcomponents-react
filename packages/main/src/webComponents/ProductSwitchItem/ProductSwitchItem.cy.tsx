import { cypressPassThroughTestsFactory } from '@/cypress/support/utils';
import { ProductSwitchItem } from './index';

describe('ProductSwitchItem', () => {
  it('Basic Test (generated)', () => {
    cy.mount(<ProductSwitchItem />);
    cy.get('[ui5-product-switch-item]').should('exist');
  });

  cypressPassThroughTestsFactory(ProductSwitchItem);
});
