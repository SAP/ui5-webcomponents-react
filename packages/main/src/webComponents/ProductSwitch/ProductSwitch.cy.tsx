import { cypressPassThroughTestsFactory } from '@/cypress/support/utils';
import { ProductSwitchItem } from '../ProductSwitchItem';
import { ProductSwitch } from './index';

describe('ProductSwitch', () => {
  it('Basic Test (generated)', () => {
    cy.mount(
      <ProductSwitch>
        <ProductSwitchItem titleText="Product Switch Item" />
      </ProductSwitch>
    );
    cy.get('[ui5-product-switch]').should('exist');
  });

  cypressPassThroughTestsFactory(ProductSwitch);
});
