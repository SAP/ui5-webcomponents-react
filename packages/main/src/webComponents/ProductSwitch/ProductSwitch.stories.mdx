import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs';
import { ProductSwitch } from './index';
import { ProductSwitchItem } from '../ProductSwitchItem';
import { DocsHeader } from '@docs/DocsHeader';
import { Footer } from '@docs/Footer';
import '@ui5/webcomponents-icons/dist/business-objects-experience.js';
import '@ui5/webcomponents-icons/dist/contacts.js';
import '@ui5/webcomponents-icons/dist/flight.js';
import '@ui5/webcomponents-icons/dist/home.js';
import '@ui5/webcomponents-icons/dist/wrench.js';

<Meta
  title="Inputs / ProductSwitch"
  component={ProductSwitch}
  subcomponents={{ ProductSwitchItem }}
  argTypes={{
    children: { control: { disable: true } },
    titleText: { description: `Defines the titleText of the <code>ProductSwitchItem</code>.` },
    icon: {
      description: `Defines the icon to be displayed as a graphical element within the <code>ProductSwitchItem</code>. <br><br> Example: <br> <pre>ProductSwitchItem icon="palette"</pre><br/><br/>See all the available icons in the <ui5-link target="_blank" href="https://openui5.hana.ondemand.com/test-resources/sap/m/demokit/iconExplorer/webapp/index.html" class="api-table-content-cell-link">Icon Explorer</ui5-link>.`
    },
    subtitleText: { description: `Defines the subtitleText of the <code>ProductSwitchItem</code>.` },
    target: {
      description: `Defines a target where the <code>targetSrc</code> content must be open. <br><br> Available options are: <ul> <li><code>_self</code></li> <li><code>_top</code></li> <li><code>_blank</code></li> <li><code>_parent</code></li> <li><code>_search</code></li> </ul>`
    },
    targetSrc: {
      description: `Defines the <code>ProductSwitchItem</code> target URI. Supports standard hyperlink behavior.`
    },
    onClick: {
      description: `Fired when the <code>ProductSwitchItem</code> is activated either with a click/tap or by using the Enter or Space key.`
    }
  }}
  args={{
    titleText: 'ProductSwitchItem',
    subtitleText: 'Customizable',
    icon: 'wrench',
    target: '_blank',
    targetSrc: 'https://sap.github.io/ui5-webcomponents-react/'
  }}
/>

<DocsHeader since="0.7.0" />

<br />

## Example

<Canvas>
  <Story name="Default">
    {(args) => {
      return (
        <ProductSwitch>
          <ProductSwitchItem {...args} />
          <ProductSwitchItem titleText="Home" subtitleText="Central Home" icon="home" />
          <ProductSwitchItem
            titleText="Analytical Cloud"
            subtitleText="Analytical Cloud"
            icon="business-objects-experience"
          />
          <ProductSwitchItem titleText="Catalog" subtitleText="Ariba" icon="contacts" />
          <ProductSwitchItem titleText="Travel & Expense" subtitleText="Concur" icon="flight" />
        </ProductSwitch>
      );
    }}
  </Story>
</Canvas>

## Properties

<ArgsTable story="." />

<Footer />
