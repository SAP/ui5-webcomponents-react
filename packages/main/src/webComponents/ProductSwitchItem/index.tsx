import '@ui5/webcomponents-fiori/dist/ProductSwitchItem.js';
import { CommonProps } from '../../interfaces/CommonProps';
import { Ui5CustomEvent } from '../../interfaces/Ui5CustomEvent';
import { Ui5DomRef } from '../../interfaces/Ui5DomRef';
import { withWebComponent } from '../../internal/withWebComponent';

interface ProductSwitchItemAttributes {
  /**
   * Defines the icon to be displayed as a graphical element within the component.
   *
   * Example:
   *
   * ProductSwitchItem icon="palette"
   *
   * See all the available icons in the <ui5-link target="_blank" href="https://sdk.openui5.org/test-resources/sap/m/demokit/iconExplorer/webapp/index.html" class="api-table-content-cell-link">Icon Explorer</ui5-link>.
   */
  icon?: string;
  /**
   * Defines the subtitle of the component.
   */
  subtitleText?: string;
  /**
   * Defines a target where the `targetSrc` content must be open.
   *
   * Available options are:
   *
   * *   `_self`
   * *   `_top`
   * *   `_blank`
   * *   `_parent`
   * *   `_search`
   */
  target?: string;
  /**
   * Defines the component target URI. Supports standard hyperlink behavior.
   */
  targetSrc?: string;
  /**
   * Defines the title of the component.
   */
  titleText?: string;
}

export interface ProductSwitchItemDomRef extends ProductSwitchItemAttributes, Ui5DomRef {}

export interface ProductSwitchItemPropTypes extends ProductSwitchItemAttributes, Omit<CommonProps, 'onClick'> {
  /**
   * Fired when the `ProductSwitchItem` is activated either with a click/tap or by using the Enter or Space key.
   */
  onClick?: (event: Ui5CustomEvent<ProductSwitchItemDomRef>) => void;
}

/**
 * The `ProductSwitchItem` web component represents the items displayed in the `ProductSwitch` web component.
 *
 * **Note:** `ProductSwitchItem` is not supported when used outside of `ProductSwitch`.
 *
 * <ui5-link href="https://sap.github.io/ui5-webcomponents/playground/components/ProductSwitch" target="_blank">UI5 Web Components Playground</ui5-link>
 */
const ProductSwitchItem = withWebComponent<ProductSwitchItemPropTypes, ProductSwitchItemDomRef>(
  'ui5-product-switch-item',
  ['icon', 'subtitleText', 'target', 'targetSrc', 'titleText'],
  [],
  [],
  ['click']
);

ProductSwitchItem.displayName = 'ProductSwitchItem';

ProductSwitchItem.defaultProps = {
  target: '_self'
};

export { ProductSwitchItem };
