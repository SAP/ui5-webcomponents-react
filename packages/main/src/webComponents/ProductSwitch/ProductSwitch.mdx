import { ArgTypesWithNote, ControlsWithNote, DocsHeader, Footer } from '@sb/components';
import SubcomponentsSection from '@sb/docs/SubcomponentsSection.md?raw';
import { Canvas, Markdown, Meta } from '@storybook/blocks';
import { ProductSwitchItem } from '../ProductSwitchItem';
import ProductSwitchItemDescription from '../ProductSwitchItem/ProductSwitchItemDescription.md?raw';
import * as ComponentStories from './ProductSwitch.stories';
import Description from './ProductSwitchDescription.md?raw';

<Meta of={ComponentStories} />

<DocsHeader since="0.7.0" />

<br />

## Example

<Canvas of={ComponentStories.Default} />

## Properties

<ControlsWithNote of={ComponentStories.Default} />

<Markdown>{Description}</Markdown>

<Markdown>{SubcomponentsSection}</Markdown>

## ProductSwitchItem

<ArgTypesWithNote of={ProductSwitchItem} />
<Markdown>{ProductSwitchItemDescription}</Markdown>

<Footer />
