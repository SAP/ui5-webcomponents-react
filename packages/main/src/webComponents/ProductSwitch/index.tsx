'use client';

import '@ui5/webcomponents-fiori/dist/ProductSwitch.js';
import { ReactNode } from 'react';
import { CommonProps } from '../../interfaces/CommonProps';
import { Ui5DomRef } from '../../interfaces/Ui5DomRef';
import { withWebComponent } from '../../internal/withWebComponent';

interface ProductSwitchAttributes {}

export interface ProductSwitchDomRef extends ProductSwitchAttributes, Ui5DomRef {}

export interface ProductSwitchPropTypes extends ProductSwitchAttributes, CommonProps {
  /**
   * Defines the items of the `ProductSwitch`.
   */
  children?: ReactNode | ReactNode[];
}

/**
 * The `ProductSwitch` is an SAP Fiori specific web component that is used in `ShellBar` and allows the user to easily switch between products.
 *
 * __Note:__ This component is a web component developed by the UI5 Web Componentsâ€™ team.
 *
 * <ui5-link href="https://sap.github.io/ui5-webcomponents/playground/components/ProductSwitch" target="_blank">UI5 Web Components Playground</ui5-link>
 */
const ProductSwitch = withWebComponent<ProductSwitchPropTypes, ProductSwitchDomRef>(
  'ui5-product-switch',
  [],
  [],
  [],
  [],
  () => import('@ui5/webcomponents-fiori/dist/ProductSwitch.js')
);

ProductSwitch.displayName = 'ProductSwitch';

export { ProductSwitch };
