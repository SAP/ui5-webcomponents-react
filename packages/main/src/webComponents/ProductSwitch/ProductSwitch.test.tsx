import { render } from '@shared/tests';
import { createCustomPropsTest } from '@shared/tests/utils';
import { ProductSwitch } from '@ui5/webcomponents-react/dist/ProductSwitch';
import { ProductSwitchItem } from '@ui5/webcomponents-react/dist/ProductSwitchItem';
import React from 'react';

describe('ProductSwitch', () => {
  test('Basic Test (generated)', () => {
    const { asFragment } = render(
      <ProductSwitch>
        <ProductSwitchItem titleText="Product Switch Item" />
      </ProductSwitch>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  createCustomPropsTest(ProductSwitch);
});
