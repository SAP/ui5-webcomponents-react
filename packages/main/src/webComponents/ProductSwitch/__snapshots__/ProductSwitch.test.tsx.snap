// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSwitch Basic Test (generated) 1`] = `
<DocumentFragment>
  <ui5-product-switch
    ui5-product-switch=""
  >
    <ui5-product-switch-item
      target="_self"
      title-text="Product Switch Item"
      ui5-product-switch-item=""
    />
  </ui5-product-switch>
</DocumentFragment>
`;
